src/tsickle.ts

```diff
@@ -472,10 +482,18 @@ class Annotator extends ClosureRewriter {
           if (exportDecl.exportClause) this.visit(exportDecl.exportClause);
         }
         if (exportDecl.moduleSpecifier) {
-          this.emit(' from');
-          this.writeModuleSpecifier(exportDecl.moduleSpecifier);
+          this.emit(` from '${this.resolveModuleSpecifier(exportDecl.moduleSpecifier)}';`);
+          const symbolPrefix = this.emitGoogRequire(exportDecl.moduleSpecifier);
+          if (exportDecl.exportClause) {
+            this.registerSymbolPrefixes(symbolPrefix, exportDecl.exportClause);
+          }
+        } else {
+          this.emit(';');
+        }
+        if (exportDecl.exportClause) {
+          // TODO(martinprobst): typedefs from export *.
```

Comment:
This code did not special case typedefs exported through `export *`, as those need to be special cased because they don't create a value, just a type. I've fixed this by implementing it.